tests/conftest.py

```diff
@@ -4361,3 +4368,218 @@ def finalizer():
+        assert (
+            "access_key" in acc_creation_resp.text
+        ), f"Did not find access_key in account creation response. Response: {acc_creation_resp.text}"
+        # Let the account propagate through the system
+        time.sleep(5)
+        access_keys = json.loads(acc_creation_resp.text)["reply"]["access_keys"][0]
+        s3_creds = {
+            "access_key_id": access_keys["access_key"],
+            "access_key": access_keys["secret_key"],
+            "endpoint": mcg_obj_session.s3_endpoint,
+            "ssl": False,
+        }
+        nsfs_obj.s3_creds = s3_creds
+        # Create a boto3 S3 resource for commmunication with the NSFS bucket
+        nsfs_s3_client = boto3.resource(
+            "s3",
+            verify=False,
+            endpoint_url=s3_creds["endpoint"],
+            aws_access_key_id=s3_creds["access_key_id"],
+            aws_secret_access_key=s3_creds["access_key"],
+        )
+        # Create a new NSFS bucket
+        # Follow this flow if the bucket should be created on top of an existing directory
+        if nsfs_obj.mount_existing_dir:
+            nsfs_obj.bucket_name = helpers.create_unique_resource_name(
+                resource_description="nsfs-bucket", resource_type="rpc"
+            )
+            new_dir_path = f"/{nsfs_obj.bucket_name}"
+            nsfs_interface_pod.exec_cmd_on_pod(
+                f"mkdir -m {nsfs_obj.existing_dir_mode} {nsfs_obj.mount_path}/{new_dir_path}"
+            )
+            rpc_bucket_creation_response = mcg_obj_session.send_rpc_query(
+                "bucket_api",
+                "create_bucket",
+                {
+                    "name": nsfs_obj.bucket_name,
+                    "namespace": {
+                        "write_resource": {
+                            "resource": nsfs_obj.nss.name,
+                            "path": new_dir_path,
+                        },
+                        "read_resources": [
+                            {"resource": nsfs_obj.nss.name, "path": new_dir_path}
+                        ],
+                    },
+                },
+            )
+            if nsfs_obj.verify_health:
+                assert (
+                    "optimal" in rpc_bucket_creation_response.text.lower()
+                ), f"RPC bucket isn't in optimal state; Full response: {rpc_bucket_creation_response.text}"
+            created_rpc_buckets.append(nsfs_obj.bucket_name)
+        # Otherwise, the new bucket will create a directory for itself
+        else:
+            nsfs_obj.bucket_name = bucket_factory(
+                s3resource=nsfs_s3_client, verify_health=nsfs_obj.verify_health
+            )[0]
+        nsfs_obj.mounted_bucket_path = f"{nsfs_obj.mount_path}/{nsfs_obj.bucket_name}"
+
+    def nsfs_bucket_factory_cleanup():
+        """
+        Delete the Pod or the DeploymentConfig
+
+        """
+        for account_email in created_accounts:
+            mcg_obj_session.send_rpc_query(
```

Comment:
Can NSFS account deletion handled as a NSFS account factory fixture, instead of deleting at end of each tests?